File: booking/src/main/java/wbms/com/sg/booking/service/BookingService.java
package wbms.com.sg.booking.service;

import wbms.com.sg.booking.dto.BookingDTO;
import wbms.com.sg.booking.dto.ComUserDTO;
import wbms.com.sg.booking.entity.Booking;

import java.util.*;
import java.time.LocalDate;
import java.time.LocalDateTime;
public interface BookingService {

  ComUserDTO getUserFromAccountsService();

  List<Booking> findAll();

  List<Booking> getBookingsByUser(String employeeId);

  Booking findById(Long id);
  // Booking updateBookingByUser(BookingDTO bookingDTO);

  List<Booking> getBookingsByIds(List<Long> idList);

  void updateBooking(BookingDTO bookingDTO);

  List<Booking> findByRescId(Long rescId, LocalDate date);

}

File: frontend/src/app/services/booking.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';
import { Booking } from '../booking/booking.model';
import { RequestService } from './request.service';
import { environment } from 'src/environment';

@Injectable()

export class BookingService {

    apiUrl: any = environment.api.booking;

  constructor(
        private http: HttpClient,
        private requestService: RequestService
    ) { }


    findAll(): Observable<any[]> {
        return this.http.get<any[]>(`${this.apiUrl}/findAll`);
    }

    getBookingsByUser(empId:string): Observable<any[]> {
        return this.http.get<any[]>(`${this.apiUrl}/getBookingsByUser/${empId}`);
    }

    findById(Id:any): Observable<any> {
        return this.http.get<any>(`${this.apiUrl}/findById/${Id}`);
    }
    
    updateBooking(booking: Booking): Observable<any[]> {
        return this.requestService.create(`/updateBooking/`, this.apiUrl, booking);
    }  

    getBookingsByIds(idList: number[]): Observable<any[]> {
        return this.http.post<any[]>(`${this.apiUrl}/getBookingsByIds`, idList);
    }

    findByRescId(rescId:any, date:any): Observable<any> {
        return this.http.get<any>(`${this.apiUrl}/findByRescId/${rescId}/${date}`);
    }
}